src/main/java/edu/kis/powp/jobs2d/drivers/monitoring/DriverMonitorDecorator.java

```diff
@@ -2,42 +2,74 @@
 
 import edu.kis.powp.jobs2d.drivers.AbstractDecorator;
 import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
+import edu.kis.powp.jobs2d.drivers.monitoring.DriverEventManager.DriverEventType;
 import edu.kis.powp.observer.Publisher;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
 
 public class DriverMonitorDecorator extends AbstractDecorator {
     private final DriverUsageMonitor monitor;
     private final Publisher movePublisher = new Publisher();
-    private static boolean monitorEnabled = true;
-    public DriverMonitorDecorator(VisitableJob2dDriver driver, DriverUsageMonitor monitor) {
+    private final DriverMonitoringConfig config;
+
+    private static boolean globalMonitorEnabled = true;
+
+    public DriverMonitorDecorator(VisitableJob2dDriver driver, DriverUsageMonitor monitor, DriverMonitoringConfig config) {
         super(driver);
-        this.monitor = monitor;
+        this.monitor = Objects.requireNonNull(monitor, "Monitor cannot be null");
+        this.config = config != null ? config : createDefaultConfig();
     }
-    public static boolean isMonitorEnabled() {
-        return monitorEnabled;
+
+    public DriverMonitorDecorator(VisitableJob2dDriver driver, DriverUsageMonitor monitor) {
+        this(driver, monitor, null);
     }
 
+    private void notifyObserversIfEnabled() {
+        if (config.isMonitorEnabled()) {
```

Comment:
if we need config only for this flag maybe apply ISP and DIP (small interfaces) or maybe it should be a responisibility of monitor to notify observers?